logstash-core/lib/logstash/elasticsearch_client.rb

```diff
@@ -39,8 +39,11 @@ def initialize(settings, logger)
       @client_args[:ssl] = ssl_options
 
       username = @settings["var.elasticsearch.username"]
-      password = @settings["var.elasticsearch.password"]
       if username
+        password = @settings["var.elasticsearch.password"]
+        if password.is_a?(LogStash::Util::Password)
```

Comment:
by doing that, we make sure that our lib wont bleed anything, if we bleed password it will be outside.